### PR TITLE
Manage sending domains

### PR DESCRIPTION
**Pull request by zeuss2011**
This is a beginning for sending domains managing

### CODE REVIEW

**Comment by darrencauthon** [comment]
Thank you very much for this work.

I'd like to import it, but I can't import it with all of the Json library attributes (```JsonProperty```) all over the objects.  The conversion of data from this library to SparkPost is handled through the ```DataMapper``` through common conventions, and the data from SparkPost to this library is (currently) done manually.

There are a few reasons, but the main one is that I don't want this library to be glued to Json.  I don't have plans on replacing Json, but I will if it becomes the right decision. :smile:

**Comment by darrencauthon** [comment]
Here is a pull request that I wrote where I fixed code that was using the Json attributes everywhere:  https://github.com/darrencauthon/csharp-sparkpost/pull/63

I did that once, and... I'm never doing it again. :smile:

**Comment by darrencauthon** [comment]
I'm not sure why the Travis tests didn't populate.  I'll look into that.

**Comment by zeuss2011** [comment]
Thank you, I understand your reasons, I will remove all json attributes and use your custom DataMapper. I'll notify you once finished.

**Comment by darrencauthon** [comment]
Awesome, thanks!

**Comment by zeuss2011** [comment]
I have removed all references to Json attributes :-)

**Comment by darrencauthon** [comment]
Did you push the branch?  I still see them.

**Comment by zeuss2011** [comment]
I'm sorry, I have just pushed to my remote branch, thanks

File: src/SparkPost/DataMapper.cs [hunk 1]
```diff
@@ -55,6 +57,29 @@ public DataMapper(string version = "v1")
...
+            var data = new Dictionary<string, object>
+            {
+                ["status"] =
+                    sendingDomain.Status != null 
+                        ? WithCommonConventions(sendingDomain.Status)
+                        : null,
+                ["dkim"] =
+                    sendingDomain.Dkim != null
+                        ? WithCommonConventions(sendingDomain.Dkim)
+                        : null
```
Comment (darrencauthon) [inline src/SparkPost/DataMapper.cs]: - [ ] I'm not sure these ternaries are necessary.

**Comment by darrencauthon** [comment]
@zeuss2011 Thanks for your contribution.  I still need to run this in my test account and add the bits to my integration tests.  Once I've done those things, I'll pop a release.  I'll have it out before the holiday weekend is over.

**Comment by darrencauthon** [comment]
@zeuss2011 Oh, I did a few minor things.  The only thing thing that should affect you is that I renamed "Get" to "Retrieve" and I dropped the "ByDomain" wording.  Because they do the lookups by domain, and nothing else, so... no need to distinguish it.  :smile:

Thanks again!

**Comment by zeuss2011** [comment]
Hi @darrencauthon, sorry for replying too late. I am sick since few days. Thank you to take my work into this project.

**Comment by darrencauthon** [comment]
It's no problem @zeuss2011 .  I'm sorry I haven't deployed the update yet, I just haven't sat down and focused on testing it for myself.  But I will soon. Thanks again!

**Comment by darrencauthon** [comment]
@zeuss2011 I'm sorry, I meant to update you on this.  Version 1.9 has been released.  I made a few renames of your methods just to stay in line with the rest of the library, but otherwise it should be ready.

Thanks again for your contribution!

**Comment by zeuss2011** [comment]
You are welcome. Thanks to take in account my contribution

**Pull request merged**